src/components/MerchantList.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import Merchant from './Merchant';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+
+const initialMerchants = [
+  { id: 0, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 1, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 2, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 3, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 4, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 5, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+];
+
+function MerchantList() {
+  const [merchants, setMerchants] = useState(initialMerchants); 
+  const [hasMore, setHasMore] = useState(true); 
+  const [loading, setLoading] = useState(false); 
+  const [isBottom, setIsBottom] = useState(false);
+  const loadMoreMerchants = () => {
+    if (loading || !hasMore) return;
+    setLoading(true);
+    setTimeout(() => {
+      const newMerchants = merchants.concat(
+        Array.from({ length: 3 }).map((_, index) => ({
+          id: merchants.length + index,
+          name: `新商家 ${merchants.length + index}`,
+          distance: (Math.random() * 10).toFixed(1),
+          costDownLimit: Math.floor(Math.random() * 100),
+          costUpLimit: Math.floor(Math.random() * 200),
+          starRate: (Math.random() * 5).toFixed(1),
+          starNumber: Math.floor(Math.random() * 100),
+        }))
+      );
+
+      setMerchants(newMerchants);
```

Comment:
Don't set new state like this.
Use updater function to set new state
for example:

```javascript
 setMerchants(prevMerchants => [...prevMerchants, ...newMerchants]);
```
Read this reference:
https://medium.com/@linyawun031/react-%E5%9C%A8%E5%AD%90-component-%E8%A7%B8%E7%99%BC%E7%88%B6-component-%E8%B3%87%E6%96%99%E6%9B%B4%E6%96%B0-%E8%AA%8D%E8%AD%98-usestate-%E7%9A%84-batch-update-%E8%88%87-updater-function-6fc13d21042b